src/jemalloc.c

```diff
@@ -1949,6 +1955,16 @@ malloc_init_hard_a0_locked(void) {
 
 	malloc_init_state = malloc_init_a0_initialized;
 
+	void *readlink_allocated = NULL;
+	size_t len = strlen(readlink_buf);
+	if (len > 0) {
+		readlink_allocated = a0ialloc(len + 1, false, true);
+	}
+	if (readlink_allocated != NULL) {
+		memcpy(readlink_allocated, readlink_buf, len + 1);
+		malloc_conf_file = readlink_allocated;
+	}
```

Comment:
Refactor a bit:
```
size_t readlink_buf_len = strlen(readlink_buf);
if (readlink_buf_len > 0) {
    void *readlink_allocated = a0ialloc(len + 1, false, true);
    memcpy(readlink_allocated, readlink_buf, len + 1);
    malloc_conf_file = readlink_allocated;
}
```

Reply:
Does `a0ialloc` guarantee non-null result? memcpy on NULL will result in UB?

Reply:
Nope you will still want to guard memcpy with the non-NULL check